luigi_pipeline/seqr_loading_optimized.py

```diff
@@ -38,11 +38,11 @@ def requires(self):
     def run(self):
         mt = hl.read_matrix_table(self.input()[0].path)
 
-        if self.remap_path: check_if_path_exists(self.remap_path, "remap_path")
-        if self.subset_path: check_if_path_exists(self.subset_path, "subset_path")
-        if self.remap_path:
+        # Note project remap path is optional and we can skip if it doesn't exist even if provided.
+        if self.remap_path and does_file_exist(self.remap_path):
```

Comment:
How does that align with this logic: https://github.com/broadinstitute/seqr-pipeline-airflow/blob/master/airflow/dags/seqr-vcf-to-es.py#L450-L452?  Should `project_remap_path` just be required?

Reply:
`project_remap_path` shouldn't be required -- only a few projects need to be remapped.  In the current set up, the remapping just wouldnt happen if we upload the file to GCS with the wrong name but the pipeline would fail saying it cannot find the sample IDs during subsetting because the subset file includes the remapped IDs when remapping is required. I agree this change matches the current logic but I think we can improve that failure message here by failing at the actual remapping step instead of moving forward only to fail during subsetting and adding `check_if_file_exists` under `if self.remap_path` has would accomplish that.

Reply:
gotcha.  even if `check_if_file_exists` is under `self.remap_path`, but the file does not exist, that code path won't be called though right? Because we've pruned the argument on the airflow side?  So we would still fail inside the subsetting.

Reply:
I guess the presumption here is that this code path (the genotypes task inside `seqr_loading_optimized`) is only ever called by airflow and is never run locally.

Reply:
Ah, maybe I misunderstood our slack conversation, I thought you were removing that logic from the airflow side: "is that something the luigi side should just handle? "

Reply:
I think we can?  Would that be the best path here?

- Remove the logic on airflow to check arguments existing on gcs.
- Hard fail here if `remap_path` doesn't exist on gcs OR the `subset_path` does not exist on GCS.

Reply:
Also...  I just added back the `check_if_file_exists` to `remap_path`.  I think we can just open a ticket for resolving the is "remap_path" optional question cause it seems pretty hairy.

Reply:
Yeah, I guess if we consider airflow is the only thing calling this and is pruning the arg then ` and does_file_exist(self.remap_path)` isn't really doing anything either since it would have had to existed for airflow to pass the optional arg in the first place?  I think the note `and we can skip if it doesn't exist even if provided.` just is not the behavior we'd want. If it's passed,  we dont want to skip the remap step if it doesnt exist. I have no idea when this would happen, someone would have to delete the correctly placed file after the task is triggered and before hail tries to read it...